### PR TITLE
Coupons: Update coupon list items following the latest design

### LINKED ISSUE #6486: Coupons: Update coupon list item following the latest design
The list items should include summary for the coupon and have a new layout:

### PR DESCRIPTION
**Pull request by itsmeichigo**
Closes: #6486 

### Description

This PR updates the layout and content of coupon list items following the latest design.
Previously we were showing only the discount type for the subtitle of the items. In this update, it's replaced with a summary line of the formatted discount amount and apply rules.

I've moved the logic for the formatted amount and apply rules from the coupon details view model to the extension of `Coupon` to make it easier to reuse. The duplicated logic will be removed in the next PR when the coupon details screen layout is updated.

There's a grey area for formatting the discount amount of unsupported discount types. In this PR I'm removing the amount for simplicity.

### Testing instructions

- Please make sure that your test store has coupons enabled and set up a few coupons.
- Launch the app, make sure that you have Coupon Management enabled in Menu > Settings > Experimental Features.
- Navigate to Menu > Coupons. Notice that the coupon items in the list are updated with new layout and display correct information for each coupon. 

### Screenshots

| Before | After |
| ----- | ----- |
|  | 

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by pmusolino** [comment]
Could I review this @itsmeichigo? 🙇

**Comment by itsmeichigo** [comment]
Oh yes please @pmusolino, I didn't want to spam you with too many PRs at a time so I was leaving it for today 😅

File: WooCommerce/Classes/Model/Coupon+Woo.swift [hunk 1]
```diff
@@ -47,6 +48,78 @@ extension Coupon {
...
+        }
+        return amountString
+    }
+
+    /// Localize content for the "Apply to" field. This takes into consideration different cases of apply rules:
+    ///    - When only specific products or categories are defined: Display "x Products" or "x Categories"
+    ///    - When specific products/categories and exceptions are defined: Display "x Products excl. y Categories" etc.
+    ///    - When both specific products and categories are defined: Display "x Products and y Categories"
+    ///    - When only exceptions are defined: Display "All excl. x Products" or "All excl. y Categories"
+    ///
```
Comment (pmusolino) [inline WooCommerce/Classes/Model/Coupon+Woo.swift:88]: Thanks for adding a comment with all the cases, really helpful!

**Review by pmusolino — APPROVED** [review]
Looks good to me 👍

**Pull request merged**